src/animated/Controller.ts

```diff
@@ -69,123 +92,131 @@ class Controller<P extends any = {}> {
           to: { ...previous, ...entry.to },
         }
       })
-      this.queue = Object.values(ops)
+      ops.forEach(op => this.queue.push(op))
     }
+
     // Sort queue, so that async calls go last
-    this.queue = this.queue.sort((a, b) => a.delay - b.delay)
+    this.queue.sort((a, b) => a.delay - b.delay)
 
     // Diff the reduced props immediately (they'll contain the from-prop and some config)
-    this.diff(props)
+    if (hasKeys(props)) this.diff(props)
+
     return this
   }
 
-  /** start(onEnd)
-   *  This function either executes a queue, if present, or starts the frameloop, which animates */
-  start(onEnd?: FinishedCallback) {
-    // If a queue is present we must excecute it
+  /** start()
+   *  This function either executes a queue, if present, or starts the frameloop, which animates.
+   *  The `useSpring` hooks never have > 1 update per call, because they call this every render. */
+  start(onEnd?: EndCallback) {
+    // If a queue is present we must execute it
     if (this.queue.length) {
-      this.idle = false
+      const { prevQueue } = this
 
       // Updates can interrupt trailing queues, in that case we just merge values
-      if (this.localQueue) {
-        this.localQueue.forEach(({ from = {}, to = {} }) => {
```

Comment:
I removed the `= {}` parts, because duck-typing is used via the `is.obj` function.